common/lib/common-definitions/src/logger.ts

```diff
@@ -15,32 +15,47 @@ export type TelemetryEventCategory = "generic" | "error" | "performance";
  * easily add fields to objects that shouldn't be logged and not realize it's going to be logged.
  * General best practice is to explicitly log the fields you care about from objects.
  */
-export type TelemetryEventPropertyType = string | number | boolean | undefined;
+export type TelemetryEventPropertyType =
+    | string
+    | number
+    | boolean
+    | undefined
+    | (string | number | boolean)[];
 
 /**
  * A property to be logged to telemetry containing both the value and a tag. Tags are generic strings that can be used
  * to mark pieces of information that should be organized or handled differently by loggers in various first or third
  * party scenarios. For example, tags are used to mark PII that should not be stored in logs.
  */
-export interface ITaggedTelemetryPropertyType {
-    value: TelemetryEventPropertyType;
+export interface ITaggedTelemetryPropertyType<TValue> {
+    value: TValue;
     tag: string;
 }
 
+export type OptionallyTaggedTelemetryProperty<T> = T | ITaggedTelemetryPropertyType<T>;
+
 /**
  * JSON-serializable properties, which will be logged with telemetry.
  */
 export interface ITelemetryProperties {
-    [index: string]: TelemetryEventPropertyType | ITaggedTelemetryPropertyType;
+    [index: string]: OptionallyTaggedTelemetryProperty<TelemetryEventPropertyType>;
 }
 
+/**
+ * Set of properties defined for the base logger to use only primitive types.
+ */
+export type TelemetryBaseEventPropertyType = string | number | boolean | undefined;
+
+export interface ITelemetryBaseProperties {
```

Comment:
I think you really should stage this through 2 major releases.  In the first one, add `ITelemetryBaseProperties` (including `null` option as I suggested above) and leave the existing `ITelemetryProperties` as-is, but probably mark it as deprecated.

Add an "Upcoming" note indicating that this `ITelemetryProperties` will be repurposed in the next major release (see instructions [here](https://github.com/microsoft/FluidFramework/wiki/Communicating-breaking-changes)). This will trigger a process on consumer's side to spend time before the following major release to migrate to the new name.

Then you can make a separate non-breaking change to add the new types with some temporary name like `ITelemetryProperties2` or `ITelemetryPropertiesExt` or `ITelemetryPropertiesInternal`, and migrate our FF code like `TelemetryLogger` etc to use the new type.

And finally, in the following major release you can actually switch to reuse the old name.

I know it sounds like a hassle, but it shouldn't be too bad, and we really have committed to our partners to try to stage breaking changes so they can be incorporated outside the actual upgrade PR.  It would be pretty disruptive to swap names in-place like this.

Reply:
I think it's simpler - we can add ITelemetryBaseProperties as an alias to ITelemetryProperties in main and start using it on base logger and add a note to breaking.md that people should switch.